atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/RateLimitingFailureHandler.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * (c) Copyright 2021 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.keyvalue.cassandra;
+
+import com.google.common.base.Suppliers;
+import java.time.Duration;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Supplier;
+
+/**
+ * Wraps a potentially expensive failure-handling computation in a memoizing supplier, so that it is only executed
+ * once per fixed duration.
+ */
+public class RateLimitingFailureHandler implements Runnable {
```

Comment:
```suggestion
public final class RateLimitingFailureHandler implements Runnable {
```

Reply:
Also, on the one hand this is "rate limiting", but I feel like the pattern is more some form of expiring or caching failure handler (yes, it limits the rate to once per hour, but still).

Reply:
And also also, this doesn't have to be a failure thing - this is more `RateLimitingRunnable` or `CachedRunnable` or `TimeExpiringMemoizedRunnable` or something.